### PR TITLE
navbar component with header carousel

### PR DESCRIPTION
**Pull request by tomekwoojcik**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
-</html>
+
+<head>
+  <meta charset="UTF-8" />
+  <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+  <title>Coffee Shop</title>
+  <link rel="preconnect" href="https://fonts.googleapis.com">
+  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+  <link href="https://fonts.googleapis.com/css2?family=Quicksand:wght@300;400;500;600;700&display=swap"
```
Comment (liftoff-project) [inline index.html]: Import your fonts directly to the highest scss file instead of here.

File: index.html [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
-    <script type="module" src="/src/main.tsx"></script>
-  </body>
-</html>
+
+<head>
+  <meta charset="UTF-8" />
+  <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+  <title>Coffee Shop</title>
+  <link rel="preconnect" href="https://fonts.googleapis.com">
```
Comment (liftoff-project) [inline index.html]: What's this?

File: index.html [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
+  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
```
Comment (liftoff-project) [inline index.html]: What's this for?

File: src/api/apiGetCategories.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import axios from "axios";
+
+export interface categoriesModel {
```
Comment (liftoff-project) [inline src/api/apiGetCategories.ts]: Name should be from the capital letter.

File: src/component/atom/SearchIcon/searchIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import { FC, useContext } from "react";
+import { NavbarMenuContext } from "../../../context/navbarMenuContext";
+
+const SearchIcon: FC = () => {
+  const { toggleSearchButton } = useContext(NavbarMenuContext);
+  return (
+    <IconButton
+      aria-label="cart"
+      onClick={() => {
+        toggleSearchButton(false);
```
Comment (liftoff-project) [inline src/component/atom/SearchIcon/searchIcon.tsx]: it is not good name of this function. Toggle is something that you do not need to set by passing true/false it should automatically do that instead. Please change that.

File: src/component/atom/cardCarousel/cardCarousel.tsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import { FC, Fragment } from "react";
+import { CardContentCustomize, TypographyCustomize } from "./cardCarouselCustomize";
+
+interface propsModel {
```
Comment (liftoff-project) [inline src/component/atom/cardCarousel/cardCarousel.tsx]: interfaces and types starts from capital letter

File: src/component/atom/cardCarousel/cardCarouselCustomize.tsx [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import { CardContent, Typography } from "@mui/material";
```
Comment (liftoff-project) [inline src/component/atom/cardCarousel/cardCarouselCustomize.tsx]: What this `atom` directory for?
I think that /src/components/card-carousel is enogh forr keeping all these files for now.

Comment (liftoff-project) [reply src/component/atom/cardCarousel/cardCarouselCustomize.tsx]: The name CardCarousel.styled matches better for this file name.

File: src/component/molecules/menuWrapper/menuWrapper.tsx [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import { Box } from "@mui/material";
+import { FC, useContext } from "react";
+import PopupsMenu from "../../organism/popupsMenu/popupsMenu";
+import { NavbarMenuContext } from "../../../context/navbarMenuContext";
+
+const MenuWrapper: FC = () => {
+    const {categories} = useContext(NavbarMenuContext)
+    return (
+        <Box>
+            <PopupsMenu objArr={categories}/>
```
Comment (liftoff-project) [inline src/component/molecules/menuWrapper/menuWrapper.tsx]: `objArr` is not a good name for this property. Change that on something that better describes.

File: src/component/molecules/navbarMenuLeft/navbarMenuLeftCustomize.tsx [hunk 9]
```diff
@@ -0,0 +1,7 @@
+import { Box } from "@mui/material";
+import styled from "styled-components";
+
+export const BoxCustomize = styled(Box)`
```
Comment (liftoff-project) [inline src/component/molecules/navbarMenuLeft/navbarMenuLeftCustomize.tsx]: It is not looking good. I mean, look. You name this in a very general way but keep in `navbar` things. 
Betters name or move it to a proper dir for general things.

File: src/component/molecules/navbarMenuRight/navbarMenuCustomize.tsx [hunk 10]
```diff
@@ -0,0 +1,12 @@
+import { Badge, Box } from "@mui/material";
+import styled from "styled-components";
+
+export const CustomizeBadge = styled(Badge)`
```
Comment (liftoff-project) [inline src/component/molecules/navbarMenuRight/navbarMenuCustomize.tsx]: Be consist. Sometimes you have `NameCustomize` and sometimes you have `CustomizeName`

File: src/component/molecules/navbarMenuRight/navbarMenuRight.tsx [hunk 11]
```diff
@@ -0,0 +1,38 @@
+import { FC, useContext } from "react";
+import PersonOutlinedIcon from "@mui/icons-material/PersonOutlined";
+import FavoriteBorderOutlinedIcon from "@mui/icons-material/FavoriteBorderOutlined";
+import ShoppingCartOutlinedIcon from "@mui/icons-material/ShoppingCartOutlined";
+import { IconButton } from "@mui/material";
+import { CustomizeBadge, BoxCustomize } from "./navbarMenuCustomize";
+import SearchIcon from "../../atom/SearchIcon/searchIcon";
+import { NavbarMenuContext } from "../../../context/navbarMenuContext";
+import TextFieldNavbar from "../../atom/textFieldNavbar/textFieldNavbar";
+interface propsModel {
```
Comment (liftoff-project) [inline src/component/molecules/navbarMenuRight/navbarMenuRight.tsx]: Interfaces and types starts from capital letter

File: src/component/molecules/popupMenu/popupMenu.tsx [hunk 12]
```diff
@@ -0,0 +1,39 @@
+import { FC, Fragment } from "react";
+import PopupState, { bindTrigger, bindMenu } from "material-ui-popup-state";
+import { Menu } from "@mui/material";
+import { categoriesModel } from "../../../api/apiGetCategories";
+import ArrowDropDownIcon from "@mui/icons-material/ArrowDropDown";
+import {
+  ButtonCustomize, MenuItemCustomize
+} from "./popupMenuCustomize";
+interface propsModel {
```
Comment (liftoff-project) [inline src/component/molecules/popupMenu/popupMenu.tsx]: Capital letter

File: src/component/molecules/popupMenu/popupMenu.tsx [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+  obj: categoriesModel;
```
Comment (liftoff-project) [inline src/component/molecules/popupMenu/popupMenu.tsx]: `obj` name is not good name here.

File: src/component/molecules/popupMenu/popupMenuCustomize.tsx
Comment (liftoff-project) [inline src/component/molecules/popupMenu/popupMenuCustomize.tsx:1]: What is molecules directory?

File: src/component/organism/boxSliderMenu/boxSliderMenu.tsx
Comment (liftoff-project) [inline src/component/organism/boxSliderMenu/boxSliderMenu.tsx:1]: What is `organism` directory

File: src/component/organism/navbar/_navbar.scss [hunk 14]
```diff
@@ -0,0 +1,18 @@
+@import "../../../variables/variables.scss";
+@import "../../molecules/navbarMenuLeft/_navbarMenu.scss";
+@import "../popupsMenu/popupsMenu";
+
+// .navbar {
```
Comment (liftoff-project) [inline src/component/organism/navbar/_navbar.scss]: Remove unused code

File: src/component/organism/popupsMenu/popupsMenu.tsx [hunk 15]
```diff
@@ -0,0 +1,26 @@
+import { FC } from "react";
+import { categoriesModel } from "../../../api/apiGetCategories";
+import PopupMenu from "../../molecules/popupMenu/popupMenu";
+import { BoxCustomize } from "./popupsMenuCustomize";
+
+interface propsModel {
+  objArr?: categoriesModel[];
```
Comment (liftoff-project) [inline src/component/organism/popupsMenu/popupsMenu.tsx]: `objArr` is not a proper name. You names that better describes what they do.

File: src/component/organism/popupsMenu/popupsMenu.tsx [hunk 16]
```diff
@@ -0,0 +1,26 @@
+import { FC } from "react";
+import { categoriesModel } from "../../../api/apiGetCategories";
+import PopupMenu from "../../molecules/popupMenu/popupMenu";
+import { BoxCustomize } from "./popupsMenuCustomize";
+
+interface propsModel {
```
Comment (liftoff-project) [inline src/component/organism/popupsMenu/popupsMenu.tsx]: Capital letter

File: src/component/pages/mainPage/mainPage.tsx
Comment (liftoff-project) [inline src/component/pages/mainPage/mainPage.tsx:1]: I would like you to keep this naming convention `name-of-the-directory`. Leave camel cases for hook and other ts/js stuff.

File: src/component/pages/mainPage/mainPage.tsx [hunk 17]
```diff
@@ -0,0 +1,15 @@
+import { FC } from "react";
+import { Navbar } from "../../organism/navbar/navbar";
+import { NavbarMenuProvider } from "../../../context/navbarMenuContext";
+import BoxSliderMenu from "../../organism/boxSliderMenu/boxSliderMenu";
+
+export const MainPage: FC = () => {
+  return (
+    <>
```
Comment (liftoff-project) [inline src/component/pages/mainPage/mainPage.tsx]: You don't need to use fragment here as you don't have more elements that the NavbarMenuProvider...

File: src/context/navbarMenuContext.tsx [hunk 18]
```diff
@@ -0,0 +1,87 @@
+import { createContext, useEffect, useReducer, useState } from "react";
+import { propsModel } from "../model/propsContextModel";
+import { apiGetCategories, categoriesModel } from "../api/apiGetCategories";
+import { REDUCER_ACTION_TYPE, initState, reducer } from "../hooks/navbarHooks";
+import { useMediaQuery } from "@mui/material";
+
+export interface contextModel {
```
Comment (liftoff-project) [inline src/context/navbarMenuContext.tsx]: Capital letter

File: src/context/navbarMenuContext.tsx [hunk 19]
```diff
@@ -0,0 +1,87 @@
...
+  stepperTextArr: stepperObjModel[];
+  toggleSearchButton: (value: boolean) => void;
+  handleNext: () => void;
```
Comment (liftoff-project) [inline src/context/navbarMenuContext.tsx]: You can use `VoidFunction` instead of `() => void`

File: src/context/navbarMenuContext.tsx [hunk 20]
```diff
@@ -0,0 +1,87 @@
...
+  handlePrevious: () => void;
+  matches: boolean;
```
Comment (liftoff-project) [inline src/context/navbarMenuContext.tsx]: It matches what? Use better name

File: src/context/navbarMenuContext.tsx [hunk 21]
```diff
@@ -0,0 +1,87 @@
...
+  handleMenuBurger: () => void;
+  categories: categoriesModel[] | undefined;
```
Comment (liftoff-project) [inline src/context/navbarMenuContext.tsx]: If a prop can be something or undefined then use optional operator maybe? `categories ?: categoriesModel[]`

File: src/context/navbarMenuContext.tsx [hunk 22]
```diff
@@ -0,0 +1,87 @@
...
+  state: {
+    navbarSearchButtonToggle: boolean;
+    activeStep: number;
+    burgerToggle: boolean;
+  };
+}
+export interface stepperObjModel {
```
Comment (liftoff-project) [inline src/context/navbarMenuContext.tsx]: Capital letter

File: src/hooks/navbarHooks.ts
Comment (liftoff-project) [inline src/hooks/navbarHooks.ts:1]: It is more like navbar reducer or state than navbarHooks. I don't see any hook here

File: src/model/propsContextModel.tsx [hunk 23]
```diff
@@ -0,0 +1,3 @@
+export interface propsModel{
```
Comment (liftoff-project) [inline src/model/propsContextModel.tsx]: Capital letter. What is this for anyway?

File: src/style/style.scss [hunk 24]
```diff
@@ -0,0 +1,33 @@
...
+    top: -5px !important;
+}
+
+.css-46bh2p-MuiCardContent-root {
+    width: 100vw;
+    padding-left: 0 !important;
+    padding-right: 0 !important;
+}
+
+// @media only screen and (max-width: 768px) {
```
Comment (liftoff-project) [inline src/style/style.scss]: Remove unused code

**Review by liftoff-project — CHANGES_REQUESTED** [review]

File: src/model/propsContextModel.tsx (hunk 23 above)
Comment (tomekwoojcik) [reply src/model/propsContextModel.tsx]: Without this interface typescript return bug.

**Review by liftoff-project — APPROVED** [review]

**Pull request merged**